### ISSUE TITLE
support dune properly, remove usage of c_names

### ISSUE THREAD

**Issue by olafhering** [bcpierce00/unison#317]
````
[   70s] + /usr/bin/mkdir /home/abuild/rpmbuild/BUILDROOT/unison-2.51.2.26a29f7-62.opt_409.1.x86_64
[   70s] + cd unison-2.51.2.26a29f7
[   70s] + dune_release_pkgs=unison
[   70s] + echo 2.51.2.26a29f7
[   70s] + tee VERSION
[   70s] 2.51.2.26a29f7
[   70s] + dune_for_release=
[   70s] + : dune_release_pkgs
[   70s] + test -n unison
[   70s] + echo unison
[   70s] + dune_for_release=--for-release-of-packages=unison
[   70s] + dune installed-libraries
[   70s] bigarray               (version: OCaml 4.09.0)
[   70s] bytes                  (version: OCaml 4.09.0)
[   70s] compiler-libs          (version: OCaml 4.09.0)
[   70s] compiler-libs.bytecomp (version: OCaml 4.09.0)
[   70s] compiler-libs.common   (version: OCaml 4.09.0)
[   70s] compiler-libs.optcomp  (version: OCaml 4.09.0)
[   70s] compiler-libs.toplevel (version: OCaml 4.09.0)
[   70s] dynlink                (version: OCaml 4.09.0)
[   70s] lablgtk2               (version: 2.18.8)
[   70s] lablgtk2.auto-init     (version: n/a)
[   70s] lablgtk2.glade         (version: n/a)
[   70s] lablgtk2.gnomecanvas   (version: n/a)
[   70s] lablgtk2.gtkspell      (version: n/a)
[   70s] lablgtk2.rsvg          (version: n/a)
[   70s] lablgtk2.sourceview2   (version: n/a)
[   70s] raw_spacetime          (version: OCaml 4.09.0)
[   70s] result                 (version: [distributed with Ocaml])
[   70s] seq                    (version: OCaml 4.09.0)
[   70s] stdlib                 (version: OCaml 4.09.0)
[   70s] str                    (version: OCaml 4.09.0)
[   70s] threads                (version: OCaml 4.09.0)
[   70s] threads.none           (version: OCaml 4.09.0)
[   70s] threads.posix          (version: OCaml 4.09.0)
[   70s] uchar                  (version: OCaml 4.09.0)
[   70s] unix                   (version: OCaml 4.09.0)
[   70s] + dune external-lib-deps --for-release-of-packages=unison @install
[   70s] Info: Creating file dune-project with this contents:
[   70s] | (lang dune 2.2)
[   70s] File "dune", line 10, characters 1-41:
[   70s] 10 |  (c_names bytearray_stubs osxsupport pty)
[   70s]       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[   70s] Error: 'c_names' was deleted in version 2.0 of the dune language. Use the
[   70s] (foreign_stubs ...) field instead.
[   70s] error: Bad exit status from /var/tmp/rpm-tmp.cw95L5 (%build)
````

**Comment by gdt** [comment]
Please actually explain what's wrong, and enough to help us sort this into "unison fails to meet specs" vs "there is some optional thing that people do and that needs help".

**Comment by glondu** [comment]
It would be the latter. Dune is an alternative build system for OCaml projects, and support for it has been added in Unison in ee8267b310d3b5ee51ccb16de2776167e50eb9d3. This report is about an error in Dune configuration files.

**Comment by gdt** [comment]
Is this PR still valid?  I just ran a dune build of unison git master on NetBSD 9 amd64 and it succeeded.

**Comment by olafhering** [comment]
It was apparently fixed silently with PR #582 via 45fb6065bc3672608f7c697c524d47fb003178ed
